src/Bootloader/CycleOrmBootloader.php

```diff
@@ -51,10 +50,12 @@ public function __construct(
     public function boot(
         Container $container,
         FinalizerInterface $finalizer,
-        EnvironmentInterface $env
     ): void {
         $finalizer->addFinalizer(
             static function () use ($container): void {
+                if ($container->hasInstance(EntityManagerInterface::class)) {
+                    $container->get(EntityManagerInterface::class)->clean();
```

Comment:
You can clean the heap here and the next condition convert to 'elseif'

Reply:
If you're talking about using the `true` flag, I think that's a bad idea.

- First, this flag is not in the `EntityManagerInterface`, but only in its implementation, which would cause problems with static analysis;
- Secondly, if the user wants to use his implementation of `EntityManagerInterface`, he will have no way of knowing that `heap` must be cleared in the `clean` method;
- Third, even putting the `$cleanHeap` flag in the interface will reveal details of the `EntityManagerInterface` implementation.